File: 백준/Silver/10773. 제로/제로.py
import sys

input = sys.stdin.read().strip().split('\n')

count = int(input[0])
arr = []

for i in range(1, count + 1):
    el = int(input[i])

    if el:
        arr.append(el)
    else:
        arr.pop()

result = sum(arr)
print(result)

File: 백준/Silver/10773. 제로/제로.js
let fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().split('\n');

const count = Number(input[0]);
const arr =  [];

for (let i = 1; i <= count; i++) {
  const el = Number(input[i]);

  if(el) {
    arr.push(el);
  } else {
    arr.pop();
  }
}

let result = arr.reduce((acc,cur) => acc + cur, 0);
console.log(result);